### PR TITLE
[Finishes #107335616] drawImageURL

### PR DESCRIPTION
**Pull request by Bjvanminnen**
replace drawImage (which required an existing image) with drawImageURL, which downloads an image, and then draws it to the canvas after finishing download.

Added some basic validation, and also tested the following manually:

Add a 200x600 image without any extra args. It should maintain aspect ratio, but scale to fit on canvas.
Same for 600x200.

For both images, also call with extra args. It should scale to the extra args provided.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
I should make sure this works with both http and http images.

File: apps/test/solutions/applab/ec_canvas.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        '});\n' +
+        'drawImageURL("nonexistent.jpg", function (success) {\n' +
+        '  setText("result2", success);\n' +
+        '});\n',
+      runBeforeClick: function (assert) {
+        // add a completion on timeout since this is a freeplay level
+        // provide time for image to load
+        testUtils.runOnAppTick(Applab, 400, function () {
+          assert(document.getElementById('result1').textContent, "true");
+          assert(document.getElementById('result1').textContent, "false");
```
Comment (davidsbailey) [inline apps/test/solutions/applab/ec_canvas.js]: result2

File: apps/test/solutions/applab/ec_canvas.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        '});\n' +
+        'drawImageURL("nonexistent.jpg", 5, 10, 15, 20, function (success) {\n' +
+        '  setText("result2", success);\n' +
+        '});\n',
+      runBeforeClick: function (assert) {
+        // add a completion on timeout since this is a freeplay level
+        // provide time for image to load
+        testUtils.runOnAppTick(Applab, 400, function () {
+          assert(document.getElementById('result1').textContent, "true");
+          assert(document.getElementById('result1').textContent, "false");
```
Comment (davidsbailey) [inline apps/test/solutions/applab/ec_canvas.js]: result2

File: apps/test/solutions/applab/ec_canvas.js (hunk 1 above)
Comment (Bjvanminnen) [inline apps/test/solutions/applab/ec_canvas.js]: wondering how this passed now...

Comment (davidsbailey) [inline apps/test/solutions/applab/ec_canvas.js]: assert.equal!

**Comment by davidsbailey** [comment]
LGTM after fixing up tests.

File: apps/test/solutions/applab/ec_canvas.js (hunk 1 above)
Comment (Bjvanminnen) [inline apps/test/solutions/applab/ec_canvas.js]: touche. nice catch

**Comment by Bjvanminnen** [comment]
Validated that I was able to do this with an http image and with an https image. Should double check this once not on localhost.

**Comment by Bjvanminnen** [comment]
@philbogle Does this pass the triage bar? Shouldn't be scary, shouldn't affect anything non-applab

**Comment by philbogle** [comment]
Let's do this Monday.  (We have been having a big problem with flaky tests so I'm nervous about anything which potentially introduce new test failures even if spurious.)

**Pull request merged**